doc/release/1.12.0-notes.rst

```diff
@@ -23,6 +23,20 @@ Future Changes
 Compatibility notes
 ===================
 
+DeprecationWarning to error
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Indexing with floats raises IndexError,
+  e.g., a[0, 0.0].
+* Indexing with non-integer array_like raises ``IndexError``,
+  e.g., ``a['1', '2']``
+* Indexing with multiple ellipsis raises ``IndexError``,
+  e.g., ``a[..., ...]``.
+* Indexing with boolean where integer expected raises ``IndexError``,
+  e.g., ``a[False:True:True]``.
```

Comment:
It's for 1.12, so I'll just remove it and you can update the notes later. In master
```
In [1]: arange(10)[False:True:True]
Out[1]: array([0])
```